tools/xdp/s2n-quic-xdp/src/task/testing.rs

```diff
@@ -0,0 +1,24 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+use core::time::Duration;
+use rand::prelude::*;
+use tokio::time;
+
+pub async fn random_delay() {
+    let delay = thread_rng().gen_range(0..100);
+    if delay > 0 {
+        let delay = Duration::from_micros(delay);
+        trace!("sleeping for {delay:?}");
+        time::sleep(delay).await;
+    }
+}
+
+/// The number of items to send through the test queues
+pub const TEST_ITEMS: usize = 10_000;
+
+/// The configured size of each test queue.
+///
+/// This value is purposefully low to more frequently trigger corner cases of
+/// queues wrapping and/or getting full.
+pub const QUEUE_SIZE: usize = 16;
```

Comment:
Should we also be testing with larger queue sizes than this? I don't expect this is close to what we would actually want the queue size to be.

Reply:
Yes this is much smaller than what the production queue sizes will be. However, I think after a certain size the implementation will behave the same. As long as you have a few primes and even numbers in the range I think we should be fine. That being said, it's not hard to add more at larger sizes if we'd like.

As mentioned in the comments it's set low enough where we're running into the edge cases more frequently so the testing time is better utilized.

Reply:
added large queue sizes in badcca6ac613850129d078ab732c39db579a9bb2